### PR TITLE
Fix the bug on ShowExecutionPreview when the node is in error state

### PR DESCRIPTION
**Pull request by ramramps**
**Issue** 
   When the node is in error state, ShowExecutionPreview was set on Nodes.

**Fix**
   The ShowExecutionPreview was set to True / False, ideally it should set to the value of ShowRunPreview. 

-  [x] @ikeough

### CODE REVIEW

**Comment by ramramps** [comment]
@ikeough  - Updated the foreach to query now.

**Comment by ramramps** [comment]
@ikeough  - had to change the show run preview context. Because, PreviewGraph should not execute when ShowRunPreview is not set. This has affected some custom node tests as well.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 1]
```diff
@@ -20,6 +20,13 @@ public class HomeWorkspaceModel : WorkspaceModel
 
         private bool graphExecuted;              
         public readonly bool VerboseLogging;
+       
+        /// <summary>
+        /// This is set from Dynamoviewmodel. This is required here otherwise 
+        /// preview graph will run under every context that affects some of the test.
+        /// Also preview graph should run only when showrunpreview is set to true.
+        /// </summary>        
+        public bool ShowRunPreview { get; set; }
```
Comment (ikeough) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Are we sure we want this to be part of the public API for HomeWorkspaceModel? Is there another way that we communicate this change from the UI without having to expose a public property? What would a 3rd party developer who is working with HomeWorkspaceModel think this does? 

Also, the comments on the property are the documentation for this level of the architecture. The model should not know that the view model exists, nor should its documentation :) This documentation needs to explain how to use this API as if the view model were non-existent.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 2]
```diff
@@ -443,7 +450,7 @@ public void GetExecutingNodes()
             var task = new PreviewGraphAsyncTask(scheduler, VerboseLogging);
                         
             //The Graph is executed and Show node execution is checked on the Debug menu
```
Comment (ikeough) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Debug menu? What debug menu? We're in the model layer :)

Comment (ramramps) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: It has to say settings menu. I will update this.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs (hunk 1 above)
Comment (ramramps) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Yes. Model should not know about ViewModel. But preview graph should run only when ShowRunPreview is checked on the settings menu. In the model layer this is always false, hence PreviewGraph would never run. The other way is set the ShowRunPreview on DynamoModel and WorkspaceModel can get the value from DynamModel. I thought this is straight forward. Please let me know if you have any other thoughts around this.

**Comment by ramramps** [comment]
@ikeough  - After the discussion, now GetExecutingNodes lives in HomeWorkSpaceModel, and it takes ShowRunPreview parameter. Now, for every change in node, NodeModified will be called on HomeWorkSpaceModel and HomeWorkSpaceViewModel. The NodeModified on Viewmodel computes the node delta state.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 3]
```diff
@@ -438,12 +434,12 @@ public virtual void OnSetNodeDeltaState(DeltaComputeStateEventArgs e)
             if (handler != null) handler(this, e);
         }
 
-        public void GetExecutingNodes()
+        public void GetExecutingNodes(bool ShowRunPreview)
```
Comment (ikeough) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: `showRunPreview`. I'm being picky about capitalization too.

**Comment by ikeough** [comment]
@ramramps Thanks very much! One small comment then LGTM!

**Comment by ramramps** [comment]
Thanks @ikeough . I have the description updated.

**Pull request merged**